synapse/handlers/profile.py

```diff
@@ -307,7 +307,11 @@ async def check_avatar_size_and_mime_type(self, mxc: str) -> bool:
         if not self.max_avatar_size and not self.allowed_avatar_mimetypes:
             return True
 
-        server_name, _, media_id = parse_and_validate_mxc_uri(mxc)
+        host, port, media_id = parse_and_validate_mxc_uri(mxc)
+        if port is not None:
+            server_name = host + ":" + str(port)
+        else:
+            server_name = host
```

Comment:
As @clokep says, please can we have a test for this.

> I am not sure how I can write the test for check_avatar_size_and_mime_type() since that requires an actual valid mxc url since we pull out metadata by the mxc url and then we need two homeserver instances, one of which runs on non-default port.

Do we actually need two homeserver instances? it looks like we just call `get_cached_remote_media`, which just checks the database. Could you just make a call to `store_cached_remote_media` before trying to set the avatar?

Reply:
Thanks for the pointer! I was able to create a unit test that passes with the change in this PR and fails without it.